src/main/java/com/hackathon03/codeguru/utils/MissingStringPositionCheck.java

```diff
@@ -0,0 +1,20 @@
+package com.hackathon03.codeguru.utils;
+
+public class MissingStringPositionCheck {
+
+    public void checkSubStringNoncompliant(String sampleString) {
+        final String sampleSubstring = "sample";
+        final int index = sampleString.lastIndexOf(sampleSubstring);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 7 does not find the substring and returns -1, the call to substring() on line 9 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string sampleString.